### PR TITLE
Set the deletion strategy and timeout in the checkpointers.

### PR DESCRIPTION
**Pull request by workingloong**
### What changes were proposed in this pull request?

Set the deletion strategy in the checkpointers of DDP, DeepSpeed, Megatron-LM

### Why are the changes needed?

Users may need to remove unused checkpoints to release the space of storage.

### Does this PR introduce any user-facing change?

Yes.

### How was this patch tested?

Test it in the example of nanogpt.

### CODE REVIEW

**Review by BalaBalaYi — COMMENTED** [review]
Need documentation for design and use of this part.

File: dlrover/python/elastic_agent/torch/training.py [hunk 1]
```diff
@@ -691,6 +697,10 @@ def _membership_changed(self, role, rdzv_handler: RendezvousHandler):
             return True
         return False
 
+    def stop_executor(self):
+        """Shutdown the executor to save the checkpoint."""
+        self._save_ckpt_executor.shutdown()
```
[the remark this answers is not in the document]
Comment (workingloong) [reply dlrover/python/elastic_agent/torch/training.py:702]: Yes.

File: dlrover/trainer/torch/flash_checkpoint/ddp.py [hunk 2]
```diff
@@ -40,6 +38,15 @@ class DdpCheckpointer(Checkpointer):
             you should set the local_shard_num as the number of all ranks.
         comm_backend (str): the communcation backend to create a process group,
             The default is the backend of general main process group.
+        max_to_keep (int): An integer, the number of the latest
```
[the remark this answers is not in the document]
Comment (workingloong) [reply dlrover/trainer/torch/flash_checkpoint/ddp.py:41]: Good idea!

**Comment by workingloong** [comment]
> Need documentation for design and use of this part.

Yeah, I have added the introduction of `CheckpointDeletionStrategy` into the `flash_checkpoint.md`.

**Review by BalaBalaYi — APPROVED** [review]
LGTM

**Pull request merged**